File: scripts/taskdownload_client.py
#!/usr/bin/env python
import sys
import rospy
from geometry_msgs.msg import Point
from tsu200.srv import TaskDownload
from tsu200.msg import Pos


def call_taskdownload(id_val):
    rospy.wait_for_service('task_download')
    try:
        svc = rospy.ServiceProxy('task_download', TaskDownload)
        resp = svc(id_val)
        return resp
    except rospy.ServiceException as e:
        rospy.logerr('Service call failed: %s', e)
        return None


if __name__ == '__main__':
    rospy.init_node('taskdownload_client')
    id_val = int(sys.argv[1]) if len(sys.argv) > 1 else 1
    resp = call_taskdownload(id_val)
    if resp:
        print('id:', resp.id)
        print('HomePos:', resp.HomePos)
        print('PosNum:', resp.PosNum)
        print('PosList:')
        for i, p in enumerate(resp.PosList):
            print('  %d: x=%.3f y=%.3f z=%.3f type=%s info=%s' % (i, p.x, p.y, p.z, p.task_type, p.info))

File: src/taskdownload_client.cpp
#include <ros/ros.h>
#include "tsu200/TaskDownload.h"
#include "tsu200/Pos.h"
#include <geometry_msgs/Point.h>
#include <cstdlib>

int main(int argc, char **argv) {
  ros::init(argc, argv, "taskdownload_client_cpp");
  ros::NodeHandle nh;
  ros::ServiceClient client = nh.serviceClient<tsu200::TaskDownload>("task_download");
  tsu200::TaskDownload srv;
  int id = 1;
  if (argc > 1) id = atoi(argv[1]);
  srv.request.id = id;
  if (client.call(srv)) {
    ROS_INFO("id=%d HomePos=(%f,%f,%f) PosNum=%d", srv.response.id, srv.response.HomePos.x, srv.response.HomePos.y, srv.response.HomePos.z, srv.response.PosNum);
    for (size_t i=0;i<srv.response.PosList.size();++i) {
      const tsu200::Pos &p = srv.response.PosList[i];
      ROS_INFO("  Pos %lu: x=%f y=%f z=%f type=%s info=%s", i, p.x, p.y, p.z, p.task_type.c_str(), p.info.c_str());
    }
  } else {
    ROS_ERROR("Failed to call service task_download");
    return 1;
  }
  return 0;
}
